test/integration/storageversion/storage_version_test.go

```diff
@@ -0,0 +1,147 @@
+	}
+	storageVersion := sv.Status.StorageVersions[0]
+
+	if !strings.HasPrefix(storageVersion.APIServerID, "kube-apiserver-") {
+		t.Errorf("apiserver ID doesn't contain kube-apiserver- prefix, has: %v", apiserverID)
+	}
+	expectecdVersion := crd.Spec.Group + "/" + crd.Spec.Version
+	if storageVersion.EncodingVersion != expectecdVersion {
+		t.Errorf("unexpected encoding version, expected: %v, got: %v", expectecdVersion, storageVersion.EncodingVersion)
+	}
+	if len(storageVersion.DecodableVersions) != 1 {
+		t.Errorf("unexpected number of decodable versions, expected 1 version, got: %v", storageVersion.DecodableVersions)
+	}
+	if storageVersion.DecodableVersions[0] != expectecdVersion {
+		t.Errorf("unexpected decodable version, expected %v, got: %v", expectecdVersion, storageVersion.DecodableVersions[0])
+	}
+
+	// add a new version v2 to the CRD and make it the storage version
+	newVersion := createdCRD.Spec.Versions[0]
+	newVersion.Name = "v2"
+	createdCRD.Spec.Versions[0].Storage = false
+	createdCRD.Spec.Versions = append(createdCRD.Spec.Versions, newVersion)
+	if _, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Update(context.TODO(), createdCRD, metav1.UpdateOptions{}); err != nil {
+		t.Fatalf("failed to update crd: %v", err)
+	}
+
+	expectedNewVersion := crd.Spec.Group + "/" + newVersion.Name
+	expectedDecodableVersions := []string{expectecdVersion, expectedNewVersion}
+	var lastErr error
+	if err := wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+		sv, err := kubeclient.InternalV1alpha1().StorageVersions().Get(context.TODO(), gr, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		storageVersion := sv.Status.StorageVersions[0]
+		if len(storageVersion.DecodableVersions) != 2 {
+			lastErr = fmt.Errorf("unexpected number of decodable versions, expected 2 versions, got: %v", sv.Status.StorageVersions)
+			return false, nil
+		}
+		if !reflect.DeepEqual(storageVersion.DecodableVersions, expectedDecodableVersions) {
+			lastErr = fmt.Errorf("unexpected decodable versions, expected %v, got: %v", expectedDecodableVersions, storageVersion.DecodableVersions)
+			return false, nil
+		}
+		if storageVersion.EncodingVersion != expectedNewVersion {
+			lastErr = fmt.Errorf("unexpected encoding version, expected: %v, got: %v", expectedNewVersion, storageVersion.EncodingVersion)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to wait for storage version update: %v, last error: %v", err, lastErr)
+	}
+
+	// cleanup
+	if err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.Name, metav1.DeleteOptions{}); err != nil {
+		t.Errorf("failed to delete crd: %v", err)
+	}
+	if err := kubeclient.InternalV1alpha1().StorageVersions().Delete(context.TODO(), gr, metav1.DeleteOptions{}); err != nil {
+		t.Errorf("failed to delete storage version: %v", err)
+	}
+}
```

Comment:
Can you add a stress test? something like
```golang
for i:=0; i<10; i++ {
   // flip the storage version of the CRD;
   // create a few CRs;
}
// create a different CRD, verifies that the CRD creation is not blocked by storage version update of the first CRD;
// verify that the storage version of the first CRD eventually stablize to the expected version.
```

Reply:
done